waspc/src/Wasp/Analyzer/TypeChecker/TypeError.hs

```diff
@@ -117,3 +97,38 @@ getWeakenErrorMessageAndCtx = getTypeCoercionErrorMessageAndCtx $
       [ concatPrefixAndText "Expected type: " (show t),
         concatPrefixAndText "Actual type:   " (show $ exprType texpr)
       ]
+
+getTypeCoercionErrorMessageAndCtx :: (Type -> TypedExpr -> String) -> TypeCoercionError -> (String, Ctx)
+getTypeCoercionErrorMessageAndCtx getUncoercableTypesMsg typeCoercionError = (fullErrorMsg, ctx)
+  where
+    (typesErrorMsg, ctxMsgs, ctx) = getUncoercableTypesMsgAndCtxMsgsAndCtx getUncoercableTypesMsg typeCoercionError
+    fullErrorMsg
+      | null ctxMsgs = typesErrorMsg
+      | otherwise = intercalate "\n\n" [typesErrorMsg, concatContextualMsgs ctxMsgs]
+
+-- | Takes a function for constructing a type coercion error message, and a `TypeCoercionError`.
+--
+-- It recursively traverses the error stack and returns a tuple containing:
+-- - The original type coercion error message
+-- - An array of contextual messages further explaining the original error
+-- - The error's context
+getUncoercableTypesMsgAndCtxMsgsAndCtx :: (Type -> TypedExpr -> String) -> TypeCoercionError -> (String, [String], Ctx)
+getUncoercableTypesMsgAndCtxMsgsAndCtx getUncoercableTypesMsg (TypeCoercionError (WithCtx ctx texpr) t reason) =
+  case reason of
+    ReasonList e -> second3 ("In list element" :) $ getFurtherMsgsAndCtx e
```

Comment:
"In list element" sounds a bit weird, since it could be the list element itself that is problematic, right?
So maybe "for list element" is better? Or just "in list"?
I wonder now if it was me who told you to change it this way :D? But if I did, I was wrong I think.

Reply:
I'll go with "In list".